.github/workflows/test-google-cloud-platform-install.yaml

```diff
@@ -175,9 +208,27 @@ jobs:
           --quiet \
           ${{ steps.vars.outputs.k8s_cluster_name }}
 
-    # TODO: fix this!
-    # When a GKE service is deleted, the associated NEG will not be garbage collected if the NEG is still referenced by a backend service. Dereference the NEG from the backend service to allow NEG deletion.
-    # When a cluster is deleted, standalone NEGs are not deleted.
+    - name: Delete the associated NEG (Network Endpoint Groups)
+      if: ${{ always() && steps.helm_install.outcome == 'success' && steps.fetch_neg.outcome == 'success' }}
```

Comment:
helm install might have failed, but we still might have web or events up, so maybe it would be better to make this dependent on if we were able to fetch negs and get some values & always try to run the fetch above too